Dockerfile.grpc

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024 by Alibaba.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Use the official Rust image as the base image
+FROM rust:latest as builder
+
+# The list of build argument with docker build --build-arg NAME=VALUE
+# Define the default commit of source code
+# ARG CDH_COMMIT=ed809e5
+ARG CDH_COMMIT=HEAD
+
+# Set the working directory inside the container
+WORKDIR /usr/src/guest-components
+
+# Clone the specific commit from the GitHub repository
+RUN apt-get update && apt-get install -y git \
+    && git clone https://github.com/confidential-containers/guest-components.git . \
+    && git checkout ${CDH_COMMIT}
+
+# Install additional build dependencies
+RUN apt-get install -y protobuf-compiler
+
+# Build and install confidential-data-hub with specific configurations
+RUN cd confidential-data-hub && make RESOURCE_PROVIDER=kbs,sev KMS_PROVIDER=aliyun,ehsm RPC=grpc
+RUN cd confidential-data-hub/hub && cargo build --bin grpc-cdh-tool --features bin,grpc
+
+
+FROM ubuntu:22.04
+
+LABEL org.opencontainers.image.source="https://github.com/inclavare-containers/attestation-agent"
+
+# TODO: gocryptfs will send log messages to the system's syslog service.
+#       And `SwitchToSyslog: Unix syslog delivery error` will occur.
+# Install ossfs, Gocryptofs and Runtime Dependencies
+RUN apt-get update && apt-get install -y wget gdebi-core fuse gocryptfs && \
```

Comment:
I suggest that we move the downloading steps of `gocryptfs` and `ossfs` to the `builder` stage and then `COPY` from that stage. The reason is the downloading stage would require `wget` and other dependencies which only is used in the downloading stage and will make the image size bigger.

BTW, I remembered that `ossfs` and `gocryptfs` can work alone in a single binary without any other `.so` or dependencies. you can double check this by `ldd` the binary to see whether non-system `.so` is involved.